### DISCUSSION TITLE
Can you draw an architecture diagram of Flet?

### DISCUSSION THREAD

**Discussion by jet10000** [flet-dev/flet#51] [General]
Can better understand the architectural design of Flet.

**Comment by FeodorFitsner** [comment]
That's a valid request. Sure, I'm going to do "How it works" article with a diagram.

**Reply by jet10000** [reply]
Can you draw a diagram first to show the relationship and mechanism, including flet viewer, fletd and python app.

**Reply by blaisep** [reply]
I would be happy to help... I'm kind of good with diagrams.

**Reply by FeodorFitsner** [reply]
Thanks for willing to help!

So far we've got this: https://flet.dev/blog/flet-mobile-strategy/#flet-approach

Wondering what could be improved.

**Reply by controlpl4n3** [reply]
@FeodorFitsner , IMHO This is a very good diagram. I would like to make some observations for you to consider.
It aligns with the popular [C4 model for software architecture](https://crashedmind.github.io/PlantUMLHitchhikersGuide/C4/C4Stdlib.html#c4plantuml-label) which consists of 4 layers:
- Context (the whole diagram is the context of a mobile device accessing a page)
- Component (In this case a client and a server)
- Container (Here the server component has two containers, they might run on different machines)
- Code (the code is in various files?)

You have the layers in the same diagram, so it might help to distinguish the layers from each other.
It might also help to use the literal name of the code file, eg. `server/server/server.go`
and any other literal name that might correspond to either the repo or a reference implementation/ (this makes the associations less abstract)

I will attach a marked up diagram
<img width="686" alt="flet-arch" src="https://user-images.githubusercontent.com/58838288/193087059-5dc5d918-4b0f-4d59-864b-73752b877770.png">

**Comment by controlpl4n3** [comment]
Another "pro tip" for writing docs is that there are typically two dimensions: practical knowledge and theoretical understanding

Explanation (Why.....)
Description (What...)
Instructions (How to...)
Reference (WTF...)
